sfepy/base/ioutils.py

```diff
@@ -386,3 +389,89 @@ def read_sparse_matrix_hdf5(filename, output_format=None):
         mtx.sort_indices()
 
     return mtx
+
+def save_to_pt(pt_file, group, data):
+    """ Save custom data to h5 file group to be restored by read_from_pt
+        allow saving lists, dicts, numpy arrays, scalars and all pickleable
+        objects
+    """
+
+    def save_value(type, data):
+        pt_file.create_array(group, 'type', np.array(type))
+        pt_file.create_array(group, 'data', data)
+
+    def save_dict(type, data):
+        pt_file.create_array(group, 'type', np.array(type))
+        data_group = pt_file.create_group(group, 'data')
+        for d in data:
+            g = pt_file.create_group(data_group, d)
+            save_to_pt(pt_file, g, data[d])
+
+    def save_list(type, data):
+        pt_file.create_array(group, 'type', np.array(type))
+        pt_file.create_array(group, 'len', len(data))
+        data_group = pt_file.create_group(group, 'data')
+        for i, d in enumerate(data):
+            g = pt_file.create_group(data_group, str(i))
+            save_to_pt(pt_file, g, d)
+
+    if isinstance(data, Struct):
+        save_dict('Struct', data.to_dict())
+
+    elif isinstance(data, dict):
+        save_dict('dict', data)
+
+    elif isinstance(data, list):
+        save_list('list', data)
+
+    elif isinstance(data, tuple):
+        save_list('tuple', data)
+
+    elif isinstance(data, (int, float, complex, nm.ndarray)):
+        save_value('raw', data)
+
+    elif isinstance(data, str):
+        save_value('str', np.array(data))
+
+    else:
+        save_value('pickle', np.array(pickle.dumps(data)))
+
+def read_from_pt(pt_file, group):
+    """ Read data from h5 file group saved by save_from_pt """
+    def load_list():
```

Comment:
BTW. using ``save_from_pt()`` might be better as it indicates  that the name refers to a function.